src/core/qgsvectorlayerrenderer.cpp

```diff
@@ -141,9 +142,9 @@ bool QgsVectorLayerRenderer::render()
   QgsRectangle requestExtent = mContext.extent();
   mRendererV2->modifyRequestExtent( requestExtent, mContext );
 
-  QgsFeatureRequest featureRequest = QgsFeatureRequest()
-                                     .setFilterRect( requestExtent )
-                                     .setSubsetOfAttributes( mAttrNames, mFields );
+  QgsFeatureRequest featureRequest = QgsFeatureRequest();
+  mContext.addExtraExpression( &featureRequest, requestExtent, mLayer );
```

Comment:
@sbrunner if I'm interpreting this correct (and QgsRenderContext::addExtraExpression) then this will be a HUGE performance hit. Instead of filtering by the requested extent on the provider side (where spatial indexing features on providers will be utilised), it will be forced to fetch all features for every request and then evaluate the costly intersects and WKT conversion expression within QGIS.

Possibly the recent changes in cfe91f82613210a38112bb2772a11c7a08a91c5b, 75712643cd393c88882e9eff6d8b34c29e18e6a2 and 5be0ee734fa233f2a3f893ddc72e8264d029f2d0 will mean that this extra expression class/changes are no longer required